app/src/test/java/com/example/countryinfo/CountriesPreviewViewModelTest.kt

```diff
@@ -0,0 +1,135 @@
+package com.example.countryinfo
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import androidx.lifecycle.Observer
+import com.apollographql.apollo.api.Response
+import com.apollographql.apollo.exception.ApolloException
+import com.example.GetCountriesQuery
+import io.reactivex.Observable
+import org.hamcrest.core.Is.`is`
+import org.junit.Assert
+import org.junit.Rule
+import org.junit.Test
+import org.mockito.Answers
+import org.mockito.ArgumentCaptor
+import org.mockito.Mockito
+import org.mockito.Mockito.`when`
+import org.mockito.Mockito.mock
+
+class CountriesPreviewViewModelTest {
+
+    @Rule
+    @JvmField
+    val instantExecutorRule = InstantTaskExecutorRule()
+
+    private val countriesApiMock = mock(ICountriesApi::class.java)
+
+    private val viewModel by lazy {
+        CountriesPreviewViewModel(countriesApiMock)
+    }
+
+    @Test
+    fun `when getCountries invoked then post default CountryDetailsViewState on success`() {
+        val mockResponse: Response<GetCountriesQuery.Data> =
+            mock(Response::class.java) as Response<GetCountriesQuery.Data>
+        val mockData =
+            mock(GetCountriesQuery.Data::class.java, Answers.RETURNS_DEEP_STUBS)
+        val mockCountryPreviewFragments = mockCountriesQueryList()
+
+        `when`(mockResponse.data).thenReturn(mockData)
+        `when`(mockData.country).thenReturn(
+            mockCountryPreviewFragments
+        )
+        `when`(countriesApiMock.getCountries()).thenReturn(
+            Observable.just(
+                mockResponse
+            )
+        )
+
+        val mockObserver = mock(Observer::class.java) as Observer<CountriesPreviewViewState>
+        viewModel.countriesLiveData.observeForever(mockObserver)
+
+        viewModel.getCountries()
```

Comment:
No need to call this explicitly, so it can be called inside `init` as it originally was.

Reply:
Done.